spec/controllers/idv_controller_spec.rb

```diff
@@ -167,30 +167,71 @@
       expect(response).to redirect_to idv_welcome_path
     end
 
-    context 'no SP context' do
+    describe 'SP for IdV requirement' do
+      let(:current_sp) { create(:service_provider) }
+      let(:ial) { 2 }
       let(:user) { build(:user, password: ControllerHelper::VALID_PASSWORD) }
 
       before do
         stub_sign_in(user)
-        session[:sp] = {}
+        if current_sp.present?
+          session[:sp] = { issuer: current_sp.issuer, ial: ial }
+        else
+          session[:sp] = {}
+        end
         allow(IdentityConfig.store).to receive(:idv_sp_required).and_return(idv_sp_required)
       end
 
-      context 'sp required' do
-        let(:idv_sp_required) { true }
+      context 'without an SP context' do
+        let(:current_sp) { nil }
 
-        it 'redirects back to the account page' do
-          get :index
+        context 'when an SP is required' do
+          let(:idv_sp_required) { true }
 
-          expect(response).to redirect_to account_url
+          it 'redirects back to the account page' do
+            get :index
+            expect(response).to redirect_to account_url
+          end
+
+          it 'begins the proofing process if the user has a profile' do
+            create(:profile, :verified, user: user)
+            get :index
+            expect(response).to redirect_to idv_welcome_url
```

Comment:
It looks like this says if an SP is required but not present, and the user already has a verified profile, they restart proofing. Is that correct? I would expect it not to proof if they're already verified.

Reply:
The profile is verified, but not active. The logic allows a user who has gone through proofing before to go through proofing again without an SP requesting IdV for them. This enables things like proofing again after resetting a password without a personal key.